sklearn/inspection/_display_estimator.py

```diff
@@ -0,0 +1,274 @@
+from collections import namedtuple
+from contextlib import closing
+from io import StringIO
+
+from .._config import config_context
+from ..base import BaseEstimator
+from ..pipeline import Pipeline
+from ..pipeline import FeatureUnion
+from ..compose import ColumnTransformer
+from ..ensemble import VotingClassifier, VotingRegressor
+
+
+def _estimator_tool_tip(estimator):
+    """Replace newlines to allow for css content: attr(...) to properly
+    display tooltips.
+    """
+    return str(estimator).replace('\n', '&#xa;')
+
+
+def _write_label_html(out, name, tool_tip):
+    """Write label to html"""
+    out.write('<div class="sk-label-container">'
+              '<div class="sk-label" sk-data-tooltip="{}">'
+              '{}</div></div>'.format(tool_tip, name))
+
+
+_EstHTMLInfo = namedtuple('_EstHTMLInfo',
+                          'type, estimators, names, name_tips')
+
+
+def _type_of_html_estimator(estimator):
+    """Generate information about how to display an estimator.
+    """
+    if isinstance(estimator, str):
+        return _EstHTMLInfo('single', estimator, estimator, estimator)
+
+    elif estimator is None:
+        return _EstHTMLInfo('single', estimator, 'None', 'None')
+
+    elif isinstance(estimator, Pipeline):
```

Comment:
Can we move this either into methods, or singledispatch, to avoid so much specialisation in the generic implementation?

Reply:
I agree it would be nice to move this out. Maybe this can be done with the current mixins... Can't hurt to try.